omod/src/main/java/org/openmrs/module/fhir2/web/servlet/FhirVersionUtils.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.web.servlet;
+
+import javax.servlet.http.HttpServletRequest;
+
+public class FhirVersionUtils {
+	
+	public enum FhirVersion {
+		R4,
+		R3,
+		ERROR
+	}
+	
+	public static Enum<FhirVersion> getFhirResourceVersion(HttpServletRequest request) {
+		String requestURI = request.getRequestURI();
+		String contextPath = request.getContextPath();
+		String prefix = contextPath + "/ws/fhir2";
+		if (requestURI.startsWith(prefix)) {
```

Comment:
@ibacher I get a NPE here when I do something like this

```java
return request.getScheme() + "://" + request.getServerName() + port + "/" + contextPath + "/ws/fhir2/"
		        + FhirVersionUtils.getFhirResourceVersion(request);
```
in `OpenmrsFhirAddressStrategy`

Reply:
Can you give me some more context for this? What does the new version of `OpenmrsFhirAddressStrategy` look like? What does the test case that you are using look like?